invite_service/proto/invite_service.proto

```diff
@@ -1,21 +1,40 @@
 syntax = "proto3";
+
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/empty.proto";
 
 service InviteService {
+  rpc get_invite_by_invite_id(InviteRequestByInviteId) returns (InviteResponse){}
   rpc get_all_invites(google.protobuf.Empty) returns (InvitesResponse){}
```

Comment:
you need to throw user in request to check user type, add new message RequestingUser